readthedocs/proxito/exceptions.py

```diff
@@ -0,0 +1,110 @@
+"""
+Exceptions for El Proxito.
+
+This module contains exceptions that may be raised in El Proxito application
+but are handled elsewhere in the Django project.
+
+--------------
+404 exceptions
+--------------
+
+Because an Http404 can be raised anywhere in the code,
+we want to add more context for targeted error handling and user-facing communication
+"""
+from django.http.response import Http404
+
+
+class ProxitoHttp404(Http404):
+
+    """General error class for proxity 404s."""
+
+
+class ProxitoProjectHttp404(ProxitoHttp404):
+
+    """Raised if a project was not found."""
+
+    def __init__(self, message, project_slug=None, proxito_path=None):
+        self.project_slug = project_slug
+        self.proxito_path = proxito_path
+        super().__init__(message)
+
+
+class ProxitoSubProjectHttp404(ProxitoProjectHttp404):
+
+    """Raised if a subproject was not found."""
+
+    def __init__(
+        self,
+        message,
+        project_slug=None,
+        proxito_path=None,
+        project=None,
+        subproject_slug=None,
+    ):
+        super().__init__(message, project_slug=project_slug, proxito_path=proxito_path)
+        self.project = project
+        self.subproject_slug = subproject_slug
+
+
+class ProxitoProjectPageHttp404(ProxitoProjectHttp404):
+
+    """Raised if a page inside an existing project was not found."""
+
+    def __init__(
+        self,
+        message,
+        project_slug=None,
+        proxito_path=None,
+        project=None,
+        subproject_slug=None,
```

Comment:
We can probably just pass the `project` and `subproject` as objects instead of passing their slugs separately

Reply:
Can you clarify this? I'm not sure what objects you mean to pass instead?

Reply:
Here you are passing the `project` as an object, and `project_slug` separately, when you can just pass `project` and get the slug from there.

Reply:
Ah okay yeah, it was to have consistency with the other exception classes so they could have the same kwarg pattern. You can always have `project_slug`, but `project` is depending on whether a Project was resolved.

I'll have a look at if it's easy to minimize the kwargs for this one.

Reply:
Fixed in the new exception classes :+1: